deltacat/exceptions.py

```diff
@@ -1,14 +1,160 @@
+    RETRYABLE_TIMEOUT_ERROR = "RetryableTimeoutError"
+    DEPENDENCY_DAFT_TRANSIENT_ERROR = "DependencyDaftTransientError"
+
+    # Validation Error
+    GENERAL_VALIDATION_ERROR = "GeneralValidationError"
+    CONTENT_TYPE_VALIDATION_ERROR = "ContentTypeValidationError"
 
 
 class NonRetryableError(Exception):
-    pass
+    is_retryable = False
+
+
+class RetryableError(Exception):
+    is_retryable = True
+
+
+# >>> example: raise DependencyRayError(ray_task="x")
+#
+# >>> __main__.DependencyRayError: Error Name: xx. Is Retryable Error: False. A dependency Ray error occurred, during ray_task: x
+class DependencyRayError(DeltaCatError, NonRetryableError):
+    error_name = DeltaCatErrorNames.DEPENDENCY_RAY_ERROR.value
+
+
+class DependencyDaftError(DeltaCatError, NonRetryableError):
+    error_name = DeltaCatErrorNames.DEPENDENCY_DAFT_ERROR.value
+
+
+class DependencyRayWorkerDiedError(DeltaCatError, RetryableError):
+    error_name = DeltaCatErrorNames.DEPENDENCY_RAY_WORKER_DIED_ERROR.value
+
+
+class DependencyRayOutOfMemoryError(DeltaCatError, RetryableError):
+    error_name = DeltaCatErrorNames.DEPENDENCY_RAY_OUT_OF_MEMORY_ERROR.value
+
+
+class DependencyRayRuntimeSetupError(DeltaCatError, RetryableError):
+    error_name = DeltaCatErrorNames.DEPENDENCY_RAY_RUNTIME_SETUP_ERROR.value
+
+
+class DependencyPyarrowError(DeltaCatError, NonRetryableError):
+    error_name = DeltaCatErrorNames.DEPENDENCY_PYARROW_ERROR.value
+
+
+class DependencyPyarrowInvalidError(DeltaCatError, NonRetryableError):
+    error_name = DeltaCatErrorNames.DEPENDENCY_PYARROW_INVALID_ERROR.value
+
+
+class DependencyPyarrowCapacityError(DeltaCatError, NonRetryableError):
+    error_name = DeltaCatErrorNames.DEPENDENCY_PYARROW_CAPACITY_ERROR.value
+
+
+class PymemcachedPutObjectError(DeltaCatError, RetryableError):
+    error_name = DeltaCatErrorNames.PYMEMCACHED_PUT_OBJECT_ERROR.value
+
+
+class GeneralValidationError(DeltaCatError, NonRetryableError):
+    error_name = DeltaCatErrorNames.GENERAL_VALIDATION_ERROR.value
+
+
+class ContentTypeValidationError(GeneralValidationError, NonRetryableError):
+    error_name = DeltaCatErrorNames.CONTENT_TYPE_VALIDATION_ERROR.value
+
+
+class GeneralStorageError(DeltaCatError, NonRetryableError):
+    error_name = DeltaCatErrorNames.GENERAL_STORAGE_ERROR.value
+
+
+class DependencyBotocoreError(DeltaCatError, NonRetryableError):
+    error_name = DeltaCatErrorNames.DEPENDENCY_BOTOCORE_ERROR.value
+
+
+class DownloadTableError(DeltaCatError, NonRetryableError):
+    error_name = DeltaCatErrorNames.DOWNLOAD_TABLE_ERROR.value
+
+
+class UploadTableError(DeltaCatError, NonRetryableError):
+    error_name = DeltaCatErrorNames.UPLOAD_TABLE_ERROR.value
+
+
+class GeneralThrottlingError(DeltaCatError, RetryableError):
+    error_name = DeltaCatErrorNames.GENERAL_THROTTLING_ERROR.value
+
+
+class RetryableUploadTableError(GeneralThrottlingError, RetryableError):
+    error_name = DeltaCatErrorNames.RETRYABLE_UPLOAD_TABLE_ERROR.value
```

Comment:
Same here. Keep only exceptions around that you are using. Otherwise, this is dead code.